### PR TITLE
feat: Add functionality for mobile-backend/client-attestation call

### PR DESCRIPTION
**Pull request by BiancaMihaila**
- update AttestationCaller interface to enable use of firebase token and jwk
- update jwk to a custom impl (JWK provided by jose4.jwt was not serializable) to enable use in the backend call
- update FirebaseClientAttestationManager to add logic for the attestation call once firebase call is successful
- update keystore manager to enable getting the public key in the required format

Resolves: DCMAW-10313

### CODE REVIEW

File: app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt [hunk 1]
```diff
@@ -33,14 +33,16 @@ class FirebaseClientAttestationManagerTest {
     fun check_success_response_from_get_attestation(): Unit = runBlocking {
         whenever(mockAppChecker.getAppCheckToken())
             .thenReturn(Result.success(AppCheckToken("Success")))
+        whenever(clientAttestationManager.getAttestation())
```
Comment (alex-bradbury-gds) [inline app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt:36]: don't think we should be mocking this....?

File: app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt [hunk 2]
```diff
@@ -50,6 +52,18 @@ class FirebaseClientAttestationManagerTest {
             (result as AttestationResponse.Failure).reason)
     }
 
+    @Test
+    fun check_failure_response_from_get_attestation() = runBlocking {
+        whenever(mockAppChecker.getAppCheckToken())
+            .thenReturn(Result.success(AppCheckToken("Success")))
+        whenever(clientAttestationManager.getAttestation())
```
Comment (alex-bradbury-gds) [inline app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt:59]: Same here, we're mocking the SUT (subject under test)

File: app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt
Comment (alex-bradbury-gds) [inline app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt:1]: I think maybe we need some tests in here mocking the caller return values

Comment (BiancaMihaila) [reply app/src/androidTest/java/uk/gov/android/authentication/integrity/FirebaseClientAttestationManagerTest.kt:1]: that's what I meant to do, but I completely missed the fact that I'm using the wrong one.. I've changed it now

File: app/src/main/java/uk/gov/android/authentication/integrity/usecase/JWK.kt [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
         )
     )
+
+    @Serializable
+    data class JsonWebKey(
+        val jwk: JsonWebKeyFormat
+    )
+
+    @Serializable
+    data class JsonWebKeyFormat(
```
Comment (ThomasIent) [inline app/src/main/java/uk/gov/android/authentication/integrity/usecase/JWK.kt:27]: Lovely, this is much nicer than what was here before 👍

**Review by alex-bradbury-gds — APPROVED** [review]

**Pull request merged**